### PR TITLE
feat: Added CI for all packages

### PR DESCRIPTION
**Pull request by GoldenJaden**
Added CI for packages.

Most jobs are made reusable so you can just construct your flow using them.

### CODE REVIEW

**Comment by ilyamore88** [comment]
image

I see only one comment. Also, would be great to add a title with a workspace name.

File: .github/actions/mutation-tests-changed-files/action.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        id: findPr
+
+      - name: Comment on successful mutation testing
+        uses: thollander/actions-comment-pull-request@v2
+        env:
+          STRYKER_DASHBOARD_API_KEY: ${{ inputs.stryker_dashboard_api_key }}
+        if: steps.changed-files.outputs.src_any_changed == 'true' && steps.run-mutation-tests.outcome == 'success' && ${{ steps.findPr.outputs.number != '' }}
+        with:
+          message: |
+            ## ✅ Mutation testing passed
```
Comment (neSpecc) [inline .github/actions/mutation-tests-changed-files/action.yml:49]:
```suggestion
            ✅ Mutation testing passed
```

File: .github/actions/mutation-tests-changed-files/action.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          pr_number: ${{ steps.findPr.outputs.number != '' && steps.findPr.outputs.number || '1'}}
+
+      - name: Comment on failed mutation testing
+        uses: thollander/actions-comment-pull-request@v2
+        env:
+          STRYKER_DASHBOARD_API_KEY: ${{ inputs.stryker_dashboard_api_key }}
+        if: steps.changed-files.outputs.src_any_changed == 'true' && steps.run-mutation-tests.outcome == 'failure' && ${{ steps.findPr.outputs.number != '' }}
+        with:
+          message: |
+            ## ❌ Mutation testing hasn't passed score threshold
```
Comment (neSpecc) [inline .github/actions/mutation-tests-changed-files/action.yml:69]:
```suggestion
            ❌ Mutation testing hasn't passed score threshold
```

File: .github/actions/mutation-tests-changed-files/action.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          pr_number: ${{ steps.findPr.outputs.number != '' && steps.findPr.outputs.number || '1'}}
+
+      - name: Comment on empty changes
+        uses: thollander/actions-comment-pull-request@v2
+        env:
+          STRYKER_DASHBOARD_API_KEY: ${{ inputs.stryker_dashboard_api_key }}
+        if: steps.changed-files.outputs.src_any_changed == 'false' && ${{ steps.findPr.outputs.number != '' }}
+        with:
+          message: |
+            ## ⏭️ No files to mutate
```
Comment (neSpecc) [inline .github/actions/mutation-tests-changed-files/action.yml:89]:
```suggestion
            ⏭️ No files to mutate
```

File: .github/actions/build/action.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: .nvmrc
+
+    - name: Setup environment
+      uses: ./.github/actions/setup
```
Comment (neSpecc) [inline .github/actions/build/action.yml:15]: maybe I'm missing something, but where is the "setup" action inside `.github/actions`?

File: .github/actions/mutation-tests-changed-files/action.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
+name: Run mutation tests for changed files only
```
Comment (neSpecc) [inline .github/actions/mutation-tests-changed-files/action.yml:1]:
```suggestion
name: Mutation tests for changed files
```

File: .github/actions/unit-tests/action.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: Run unit test
```
Comment (neSpecc) [inline .github/actions/unit-tests/action.yml:1]:
```suggestion
name: Unit tests
```

File: .github/actions/unit-tests/action.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    # Find current PR's number
+    - uses: jwalton/gh-find-current-pr@v1
+      id: findPr
+
+    - name: Run unit tests
+      uses: ArtiomTr/jest-coverage-report-action@v2
+      with:
+        working-directory: ./packages/${{ inputs.package-name }}
+        test-script: yarn workspace @editorjs/${{ inputs.package-name }} test
```
Comment (neSpecc) [inline .github/actions/unit-tests/action.yml:23]: I'm not sure that it folder name always equals package name.

For example, in playground they are different:

```
/packages/playground
```

```
@editorjs/document-playground
```

File: .github/actions/build/action.yml (hunk 4 above)
Comment (ilyamore88) [reply .github/actions/build/action.yml:15]: It's already in the main branch

File: .github/actions/unit-tests/action.yml (hunk 7 above)
Comment (GoldenJaden) [reply .github/actions/unit-tests/action.yml:23]: would variable's name "package-folder-name" be appropriate?

Comment (ilyamore88) [reply .github/actions/unit-tests/action.yml:23]: Agree with that. They might be different. Also, I think it's better to pass full package name like `@editorjs/document-playground`.

Comment (GoldenJaden) [reply .github/actions/unit-tests/action.yml:23]: Then we would need 2 different vars, like "package-folder-name" and "package-name", cause in some places you need the folder's name without "@editorjs". Is it worth it? 
Or it doesn't always come with @editorjs prefix?

File: .github/actions/mutation-tests-changed-files/action.yml (hunk 1 above)
Comment (GoldenJaden) [reply .github/actions/mutation-tests-changed-files/action.yml:49]: Why? It just makes the line bold

Comment (neSpecc) [reply .github/actions/mutation-tests-changed-files/action.yml:49]: it makes a big header, but it is actually a simple message, like regular comment. Emoji is used for high-speed determination, but big headers just looks messy.

File: .github/actions/unit-tests/action.yml (hunk 7 above)
Comment (neSpecc) [reply .github/actions/unit-tests/action.yml:23]: The `./packages/` path could be also unhardcoded I guess

Comment (neSpecc) [reply .github/actions/unit-tests/action.yml:23]: >in some places you need the folder's name without "@editorjs"

it is not a folder, but a package name, so it alway will include `@editorjs/...`

Comment (neSpecc) [reply .github/actions/unit-tests/action.yml:23]: You need 2 vars, like `working-directory` and `package-name`

Comment (GoldenJaden) [reply .github/actions/unit-tests/action.yml:23]: So, have 2 inputs like:
`package-path` (./package/name_of_your_package_folder)
and
`package-name`? (@editorjs/name_of_your_package)

File: .github/actions/mutation-tests-all-files/action.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+name: Run mutation tests for all files
```
Comment (neSpecc) [inline .github/actions/mutation-tests-all-files/action.yml:1]:
```suggestion
name: Mutation tests for all files
```

File: .github/workflows/dom-adapters.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
+name: Dom-adapters package workflow runner
```
Comment (neSpecc) [inline .github/workflows/dom-adapters.yml:1]:
```suggestion
name: Dom-adapters check
```

File: .github/workflows/model.yml [hunk 10]
```diff
@@ -1,118 +1,55 @@
 name: Model package workflow runner
```
Comment (neSpecc) [inline .github/workflows/model.yml:1]:
```suggestion
name: Model check
```

File: .github/workflows/playground.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
+name: Playground package workflow runner
```
Comment (neSpecc) [inline .github/workflows/playground.yml:1]:
```suggestion
name: Playground check
```

**Review by neSpecc — APPROVED** [review]

**Review by gohabereg — APPROVED** [review]

**Review by ilyamore88 — APPROVED** [review]

**Pull request merged**